restapi/src/main/java/io/stargate/web/resources/v2/RowsResource.java

```diff
@@ -363,14 +354,17 @@ private Object getRows(
   }
 
   private List<Where<?>> buildWhereForPath(Table tableMetadata, List<PathSegment> path) {
-    if (tableMetadata.partitionKeyColumns().size() > path.size()) {
+    List<Column> keys = tableMetadata.primaryKeyColumns();
+    boolean notAllPartitionKeys = path.size() < tableMetadata.partitionKeyColumns().size();
+    boolean tooManyValues = path.size() > keys.size();
+    if (tooManyValues || notAllPartitionKeys) {
```

Comment:
@dougwettlaufer as requested I have added checks here as well.
is this valid?

Reply:
Thanks @tlasica . I also have a ticket to refactor so I'll be able to use roughly the same methods across v1 and v2

Reply:
+1